### PR TITLE
Add autocomplete function and cmd arg type for global variables

### LINKED ISSUE #1496: avgt should specify var_name as VAR_TYPE, not STRING
https://github.com/rizinorg/rizin/blob/dev/librz/core/cmd_descs/cmd_analysis.yaml#L444

Just so i don't forget.

### LINKED ISSUE #1551: Autocomplete global variable names
**Is your feature request related to a problem? Please describe.**
Global variables are not autocompleted in rzshell.

**Describe the solution you'd like**
Whenever I need to reference a global variable by name, the shell should suggest available names when I press TAB.

**Describe alternatives you've considered**
None

**Additional context**
We should add a new `RzCmdArgType` in rz_cmd.h for global variables and use that new type in cmd_analysis.yaml for all the `avg` subcommands. Also, add a new case in `autocmplt_cmd_arg` for the global variable arg type.

### PR DESCRIPTION
**Pull request by DMaroo**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [x] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Now global variables support autocompletion (when `cfg.oldshell.autocompletion` is set to `false`).

**Test plan**

Add tests once the current changes are approved and reviewed.

**Closing issues**

Closes #1551
Closes #1496

### CODE REVIEW

File: librz/core/cautocmpl.c [hunk 1]
```diff
@@ -276,6 +276,18 @@ static void autocmplt_cmd_arg_any_type(RzCore *core, RzLineNSCompletionResult *r
 	rz_list_free(list);
 }
 
+static void autocmplt_cmd_arg_global_var(RzCore *core, RzLineNSCompletionResult *res, const char *s, size_t len) {
+	RzAnalysisVarGlobal *glob;
+	RzListIter *iter;
+	RzList *list = rz_analysis_var_global_get_all(core->analysis);
```
Comment (ret2libc) [inline librz/core/cautocmpl.c:282]: free the list

**Review by ret2libc — CHANGES_REQUESTED** [review]

**Comment by DMaroo** [comment]
Where should I add tests for this?

**Review by ret2libc — APPROVED** [review]

**Comment by ret2libc** [comment]
> Where should I add tests for this?

You could look at `test_autocmplt.c` and see how test_autocmplt_fcn or others are done. In the case of global variables should be simpler because you don't even need a real file+RzCore but you can just open a malloc://10 and add global variables there.

**Review by wargio — APPROVED** [review]

**Review by XVilka — APPROVED** [review]

File: test/unit/test_autocmplt.c [hunk 2]
```diff
@@ -400,6 +400,47 @@ static bool test_autocmplt_seek(void) {
...
+
+	RzLineBuffer *buf = &core->cons->line->buffer;
+	const char *s = "avg ";
+	strcpy(buf->data, s);
+	buf->length = strlen(s);
+	buf->index = buf->length;
+	RzLineNSCompletionResult *r = rz_core_autocomplete_rzshell(core, buf, RZ_LINE_PROMPT_DEFAULT);
+
+	rz_config_set_b(core->config, "cfg.oldshell.autocompletion", false); // for autocompletion to work
+
```
Comment (ret2libc) [inline test/unit/test_autocmplt.c]:
```suggestion
```

**Pull request merged**